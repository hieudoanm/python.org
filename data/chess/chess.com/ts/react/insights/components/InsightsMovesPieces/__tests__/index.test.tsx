import { render } from '@testing-library/react';
import { Insights } from '@web/services/chess/chess.dto';
import { mockResizeObserver } from '@web/utils/mock-resize-observer';
import { InsightsMovesPieces } from '..';

jest.mock('next/navigation', () => ({
  useRouter: jest.fn().mockReturnValue({
    asPath: '',
    events: { on: jest.fn(), off: jest.fn() },
  }),
  usePathname: jest.fn().mockReturnValue(''),
  useSearchParams: jest.fn().mockReturnValue(new URLSearchParams()),
}));

describe('InsightsMovesPieces', () => {
  beforeEach(() => {
    mockResizeObserver();
  });

  it('to match snapshot', () => {
    const { container } = render(
      <InsightsMovesPieces insights={{} as Insights} />
    );
    expect(container).toMatchSnapshot();
  });
});
