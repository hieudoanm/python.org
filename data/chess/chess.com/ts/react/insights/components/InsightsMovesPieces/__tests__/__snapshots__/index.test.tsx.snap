// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsightsMovesPieces to match snapshot 1`] = `
<div>
  <div
    class="card border-base-content border shadow"
  >
    <div
      class="border-base-content border-b px-8 py-4"
    >
      <div
        class="text-base md:text-lg lg:text-xl"
      >
        Pieces
      </div>
    </div>
    <div
      class="card-body"
    >
      <p
        class="text-base font-bold md:text-lg"
      >
        Moves per piece
      </p>
      <div
        class="aspect-video"
      >
        <div
          class="recharts-responsive-container"
          style="width: 100%; height: 100%; min-width: 0;"
        />
      </div>
    </div>
  </div>
</div>
`;
